### PR TITLE
Refactor isPerformanceTrackingOn into performance-impl

### PR DESCRIPTION
**Pull request by muxin**

### CODE REVIEW

**Comment by muxin** [comment]
`installPerformanceService(window)` seems to be appearing in a lot of test code but I doubt whether it's necessary.

File: test/functional/test-performance.js [hunk 1]
```diff
@@ -269,37 +307,45 @@ describe('performance', () => {
 
         expect(perf.events_.length).to.equal(2);
 
-        perf.coreServicesAvailable();
-        perf.flushQueuedTicks_();
-        perf.flush();
+        // const promise = perf.coreServicesAvailable();
```
Comment (lannka) [inline test/functional/test-performance.js]: whoops

Comment (muxin) [inline test/functional/test-performance.js]: Whoops, fixed

**Comment by muxin** [comment]
@erwinmombay I am just curious why we have those `installPerformanceService(window)` in the tests previously?

**Comment by zhouyx** [comment]
I remembered adding `installPerformanceService(window)` to test once. because I use the `performanceFor(win)` during constructor.  Will get an error during and if I don't include `installPerformanceService(window)`. Quite surprised they still work

**Comment by muxin** [comment]
It seems only `amp-access`, `amp-inabox`, `style-installer` and `amp.js` is using `performance` class. Tests for any other class should not be needing `installPerformanceService(window)`.

**Comment by muxin** [comment]
@zhouyx Maybe you are using `performanceFor(win)` in a constructor for a class but it's removed later?

**Comment by erwinmombay** [comment]
not sure, i also remember when the `getService` optimizations where introduced that you had to make sure services where installed early even though it wasn't directly in use in the System Under Test

**Comment by zhouyx** [comment]
@muxin yes

**Pull request merged**